src/main/java/org/example/Cases.java

```diff
@@ -9,127 +9,141 @@
-                        "        id\n" +
-                        "        enrichedString\n" +
-                        "    }\n" +
-                        "}")
-                .dataLoaderRegistry(dataLoaderRegistry)
-                .build());
-
-        // I don't really care about the result, all that matters is how many times the enrichmentService was called
-        // but you can have a look at executionResult.getData()
-    }
-
-    public static void subscribingToFlux(EnrichmentService enrichmentService) {
-        var executionResult = getGraphQl(enrichmentService).execute(ExecutionInput.newExecutionInput()
-                .query("subscription sub {\n" +
-                        "    flux {\n" +
-                        "        id\n" +
-                        "        enrichedString\n" +
-                        "    }\n" +
-                        "}")
-                .dataLoaderRegistry(dataLoaderRegistry)
-                .build());
-
-        Flux.from(executionResult.getData())
-                .collectList()
-                .block();
-        // of course I wouldn't block, but instead return the flux as server-sent events.
-        // again, I don't care about the data, only how many times the enrichmentService was called, which I assert in the test
-    }
-
-    public static Flux<List<Person>> subscribingToWindowedFlux(EnrichmentService enrichmentService) {
-        var executionResult = getGraphQl(enrichmentService).execute(ExecutionInput.newExecutionInput()
-                .query("subscription sub {\n" +
-                        "    fluxWindowed {\n" +
-                        "        id\n" +
-                        "        enrichedString\n" +
-                        "    }\n" +
-                        "}")
-                .dataLoaderRegistry(dataLoaderRegistry)
-                .build());
-
-        Publisher<ExecutionResult> executionResults = executionResult.getData();
-        var res = Flux.from(executionResults)
-                .map(executionRes -> {
-                    Map<String, List<Person>> singleWindowResults = executionRes.getData();
-                    List<Person> peopleWindow = singleWindowResults.get("fluxWindowed");
-                    return peopleWindow;
-                });
-        return res;
-    }
+	public static final String ENRICHMENT_DATA_LOADER = "enrichmentDataLoader";
+	private static final DataLoaderRegistry dataLoaderRegistry = new DataLoaderRegistry();
+
+	private static GraphQL getGraphQl(EnrichmentService enrichmentService) {
+		try {
+			var typeDefinitionRegistry =
+				new SchemaParser()
+					.parse(Resources.toString(Resources.getResource("schema.graphqls"), StandardCharsets.UTF_8));
+
+			DataFetcher<CompletableFuture<List<Person>>> listDataFetcher =
+				dataFetchingEnvironment -> {
+					// replacing Source.getPersonList();
+					List<Person> personList = IntStream.range(1, 100).boxed().map(Person::new).collect(Collectors.toList());
```

Comment:
you can't use this approach (it is mutating the input List<Person> directly)... you may want to move the enriching logic in the  service by passing directly the List<Person> and modify it there directly